### PR TITLE
added a logger for last execution date

### PR DESCRIPTION
**Pull request by asad-rafter**

### CODE REVIEW

File: cartridges/int_cloudinary/cartridge/scripts/jobs/steps/cldUploadContentLibraryAssets.js [hunk 1]
```diff
@@ -130,10 +131,14 @@ module.exports.Start = function (args) {
...
         }
 
+        var runTime = new Date();
         Transaction.wrap(function () {
-            var runTime = new Date();
             currentSite.preferences.custom.CLDContentLibraryJobLastExecutionDate = runTime;
         });
+        var CLDContentLibraryJobLastExecutionDate = currentSite.preferences.custom.CLDContentLibraryJobLastExecutionDate ? currentSite.preferences.custom.CLDContentLibraryJobLastExecutionDate.toString() : currentSite.preferences.custom.CLDContentLibraryJobLastExecutionDate;
+        if (runTime.toString() !== CLDContentLibraryJobLastExecutionDate) {
+            jobLogger.warn('Unable to update date, date should be : {0}', runTime.toString());
```
Comment (yuval-cloudinary) [inline cartridges/int_cloudinary/cartridge/scripts/jobs/steps/cldUploadContentLibraryAssets.js:140]: Please update the message to "Unable to update Job Last Execution Date in Custom Preferences..., date should be..."

**Review by yuval-cloudinary — CHANGES_REQUESTED** [review]

File: cartridges/int_cloudinary/cartridge/scripts/jobs/steps/cldUploadContentLibraryAssets.js (hunk 1 above)
Comment (yuval-cloudinary) [reply cartridges/int_cloudinary/cartridge/scripts/jobs/steps/cldUploadContentLibraryAssets.js]: @asad-rafter , I should have been clearer, apologizes. The purpose of the message is to let the user know where the issue is.
"Unable to update the job last execution timestamp in Custom Preferences->Cloudinary Jobs Configurations field: Content Library Job Last Execution Date"

Also, please add this validation to the other delta updates:
Update Product Feeds Job Last Execution Date
Catalog Content Job Last Execution Date

File: cartridges/int_cloudinary/cartridge/scripts/jobs/steps/cldUploadContentLibraryAssets.js [hunk 2]
```diff
@@ -78,7 +78,8 @@ module.exports.Start = function (args) {
                         continue;
                     }
 
-                    assetPublicID = jobStepHelpers.getAssetRelURL(args.file.toString());
+                    const file = args && args.file ? args.file.toString() : '';
```
Comment (yuval-cloudinary) [inline cartridges/int_cloudinary/cartridge/scripts/jobs/steps/cldUploadContentLibraryAssets.js:81]: you can define file as
`args?.file?.toString() || '';`

**Review by yuval-cloudinary — CHANGES_REQUESTED** [review]

File: cartridges/int_cloudinary/cartridge/scripts/jobs/steps/cldUploadCatalogAssets.js [hunk 3]
```diff
@@ -130,10 +130,14 @@ module.exports.Start = function (args) {
             }
         }
 
+        var runTime = new Date();
         Transaction.wrap(function () {
-            var runTime = new Date();
             Site.current.preferences.custom.CLDCatalogContentJobLastExecutionDate = runTime;
```
[the remark this answers is not in the document]
Comment (asad-rafter) [reply cartridges/int_cloudinary/cartridge/scripts/jobs/steps/cldUploadCatalogAssets.js:135]: The both do the same, I don't think there is an issue but if you think it should be consistent we can change that, no worries

File: cartridges/int_cloudinary/cartridge/scripts/jobs/steps/cldUploadContentLibraryAssets.js (hunk 2 above)
Comment (asad-rafter) [reply cartridges/int_cloudinary/cartridge/scripts/jobs/steps/cldUploadContentLibraryAssets.js:81]: "?" doesn't work in SFCC

Comment (yuval-cloudinary) [reply cartridges/int_cloudinary/cartridge/scripts/jobs/steps/cldUploadContentLibraryAssets.js:81]: thanks, good to know

**Review by yuval-cloudinary — APPROVED** [review]

**Pull request merged**